app/main.py

```diff
@@ -1,3 +1,52 @@
 class Distance:
-    # Write your code here
-    pass
+    def __init__(self, km: int | float) -> None:
+        self.km = km
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __add__(self, other: int | float) -> "Distance":
+        if not isinstance(other, Distance):
+            return Distance(self.km + other)
+        return Distance(self.km + other.km)
+
+    def __iadd__(self, other: int | float) -> "Distance":
```

Comment:
```suggestion
    def __iadd__(self, other: int | float) -> Distance:
```

Reply:
change evrywhere

Reply:
if i do i get an error NameError: name 'Distance' is not defined. Did you mean: 'isinstance'?

Reply:
Unreslved reference "Distance"

Reply:
add `from __future__ import annotations` in the beginning of file